packages/hash/api/src/graphql/typeDefs/ontology/entity-type.typedef.ts

```diff
@@ -4,7 +4,66 @@ export const entityTypeTypedef = gql`
   scalar EntityType
   scalar EntityTypeWithoutId
 
-  type PersistedEntityType {
+  interface PersistedEntityTypeInterface {
+    # These fields are repeated everywhere they're used because
+    # (a) GQL requires it - https://github.com/graphql/graphql-spec/issues/533
+    # (b) string interpolation breaks the code generator's introspection
+    #
+    # Could maybe use a custom schema loader to parse it ourselves:
+    # https://www.graphql-code-generator.com/docs/getting-started/schema-field#custom-schema-loader
+    #
+    # For now, _COPY ANY CHANGES_ from here to any type that 'implements PersistedPropertyTypeInterface'
+
+    """
+    The specific versioned URI of the entity type
+    """
+    entityTypeVersionedUri: String!
+    """
+    The user who created the entity type
+    """
+    accountId: ID!
+    """
+    The entity type
+    """
+    entityType: EntityType!
+  }
+
+  type PersistedEntityType implements PersistedEntityTypeInterface {
+    # INTERFACE FIELDS BEGIN #
+    """
+    The specific versioned URI of the entity type
+    """
+    entityTypeVersionedUri: String!
+    """
+    The user who created the entity type
+    """
+    accountId: ID!
+    """
+    The entity type
+    """
+    entityType: EntityType!
+    # INTERFACE FIELDS END #
+  }
+
+  type EntityTypeSubgraph implements PersistedEntityTypeInterface {
```

Comment:
We're using `EntityTypeRootedSubgraph` in the Rust code right?

Appreciate it's a bit wordy but I think `EntityTypeSubgraph` _could_ be a little misleading into thinking it's a subgraph with only entity types in it. Rather than one rooted at an entity type

Reply:
> We're using EntityTypeRootedSubgraph in the Rust code right?

Yes, we do, I'll change it

Reply:
Applied in https://github.com/hashintel/hash/pull/1062/commits/533f32d049be71f86a0978939c3a83add8ca6bc9